home/code_setup.py

```diff
@@ -0,0 +1,237 @@
+from __future__ import annotations
+
+import ipywidgets as ipw
+import pandas as pd
```

Comment:
Pandas is a very heavy dependency. Unless it's really needed we should avoid it here.

(I plan to remove pandas dependency in AWB in near future, and we don't depend on it anywhere else in the stack)

Reply:
FYI we depend on it at least in the QE app for the job history list. Between that and this code table, we've been discussing introducing a generalized pandas table component.

That said, let me think how crucial is pandas to the widget. 

Also, maybe we do want to move this PR to AWB, but then still need to think where the notebook goes.

Reply:
Yeah, we need to think more where to put it, perhaps the most expedient way is to put it to QeApp first. I don't think it belongs to AWB, since we will soon be converting AWB into a library-only (i.e. just widgets, no apps) so it shouldn't contain any notebooks.

Reply:
Okay, so @superstar54 and I are discussing this. We agree with you in the sense that it would be a great component of the Control page. So, @yakutovicha, status? Could we help getting the Control page integrated, then follow up with integrating this code management there (if you don't already have something for it)?

Reply:
Pandas dependency removed. Performance appears unaffected, though I guess this scales not as well as pandas. But I suppose that would be an issue when one's database exceeds 100 codes? Not sure how realistic this is.

Reply:
> Okay, so @superstar54 and I are discussing this. We agree with you in the sense that it would be a great component of the Control page. So, @yakutovicha, status? Could we help getting the Control page integrated, then follow up with integrating this code management there (if you don't already have something for it)?

@edan-bainglass I didn't finish it, but I don't think there were many things to finish. I can "revive" the PR tomorrow and try to get it ready ASAP.